src/Atmos/Model/refstate.jl

```diff
@@ -0,0 +1,74 @@
+### Reference state
+
+abstract type AbstractReferenceState{T} end
+struct ReferenceState{T} <: AbstractReferenceState{T}
+  T_ground::T
+  P_ground::T
+  q_tot_ground::T
+end
+
+vars_state(m::AbstractReferenceState    , DT) = @vars()
+vars_gradient(m::AbstractReferenceState , DT) = @vars()
+vars_diffusive(m::AbstractReferenceState, DT) = @vars()
+vars_aux(m::AbstractReferenceState      , DT) = @vars(ρ::DT, p::DT, T::DT)
+
+"""
+    column_integration_kernel!(m::ReferenceState, am::AtmosModel, state::Vars, aux::Vars)
+
+Column-wise integral kernel to compute reference state fields:
+  - ρ density
+  - p presure
+  - T temperature
+
+For accuracy, the logarithm of the expression is in the integrand, which must be exponentiated
+after computing the integral.
+"""
+function column_integration_kernel!(m::ReferenceState, am::AtmosModel, state::Vars, aux::Vars)
```

Comment:
@simonbyrne Here's an example of a kernel that will need to be called at initialization. This answers @tapios request to add reference states as laid out in 5.4 of the design docs.

I'll add a `NoReferenceState` to use as default so that we can adapt to the other existing cases, but we'll need a test for when a non-`NoReferenceState` is used. I'll add such a test once this kernel call is supported.